crates/pallet-subspace/src/lib.rs

```diff
@@ -475,13 +476,11 @@ pub mod pallet {
         /// This extrinsic must be called unsigned and it is expected that only block authors will
         /// call it (validated in `ValidateUnsigned`).
         #[pallet::weight((<T as Config>::WeightInfo::store_root_block(), DispatchClass::Mandatory, Pays::No))]
-        pub fn store_root_block(
-            origin: OriginFor<T>,
-            root_block: RootBlock,
-        ) -> DispatchResultWithPostInfo {
+        pub fn store_root_block(origin: OriginFor<T>, root_block: RootBlock) -> DispatchResult {
             ensure_none(origin)?;
-
-            Self::do_store_root_block(root_block)
+            RecordsRoot::<T>::insert(root_block.segment_index(), root_block.records_root());
+            Self::deposit_event(Event::RootBlockStored(root_block));
+            Ok(())
```

Comment:
The reason for having implementation separately was that IDE can actually understand what is going on there and can highlight errors and do autocomplete. I would prefer to leave it that way just for that.

Reply:
Not an IDE user, I thought the procedural macro already makes IDE happy.

Reply:
It does, to a degree, but it is still far from perfect unfortunately.